django_airavata/apps/workspace/static/django_airavata_workspace/js/components/storage/storage-edit/UserStorageTextEditViewer.vue

```diff
@@ -0,0 +1,87 @@
+<template>
+  <div>
+    <div class="user-storage-file-edit-viewer-status">
+      <div class="user-storage-file-edit-viewer-status-message">
+        <span v-if="saved">All the changes are saved.</span>
+        <span v-if="!saved">Changes are not saved.</span>
+      </div>
+      <div class="user-storage-file-edit-viewer-status-actions">
+        <user-storage-download-button :file="file"/>
+        <b-button
+          :disabled="saved"
+          @click="fileContentChanged"
+        >Save
+        </b-button>
+      </div>
+    </div>
+    <div style="width: 100%;" id="code">
+    </div>
+  </div>
+</template>
+
+<script>
+  import CodeMirror from 'codemirror'
+  import 'codemirror/lib/codemirror.css'
+  import 'codemirror/theme/abcdef.css'
+  import {utils} from "django-airavata-api";
+  import UserStorageDownloadButton from "./UserStorageDownloadButton";
+
+  export default {
+    name: "user-storage-file-edit-viewer",
+    props: {
+      file: {
+        required: true
+      }
+    },
+    components: {
+      UserStorageDownloadButton: UserStorageDownloadButton
+    },
+    data() {
+      return {
+        fileContent: "",
+        saved: true,
+        editor: null
+      };
+    },
+    mounted() {
+      this.setFileContent();
```

Comment:
Make sure that the CodeMirror instance is torn down when the component is [destroyed](https://vuejs.org/v2/api/#destroyed). See for example how Vuejs can be used to wrap a jQuery component: https://vuejs.org/v2/examples/select2.html